extension/encoding/jsonencodingextension/metadata.yaml

```
@@ -0,0 +1,9 @@
+type: jsonencoding
```

```suggestion
type: json_log_encoding
```

Also let's call the go module jsonlogencodingextension

---

Why are we limiting this encoding extension to logs only? Wouldn't it make sense to have a single extension that (in future) supports encoding of all signals - logs, metrics, traces? Similar to how we don't have separate `otlplogs` receiver, but a single `otlp` receiver for logs, metrics and traces.

---

Oh I think I misunderstood the purpose of this encoding extension. I assumed it is an "OTLP JSON" encoding extension, whereas it is an extension to encode/decode JSON log body, right?